src/pf-dropdown/pf-dropdown.component.js

```diff
@@ -0,0 +1,172 @@
+ *        <li class="divider"></li>
+ *        <li><a href="#">Item 5</a></li>
+ *      </ul>
+ *      </div>
+ *  </pf-dropdown>
+ */
+
+export class PfDropdown extends HTMLElement {
+
+  /**
+   * Called when an instance was inserted into the document
+   */
+  attachedCallback() {
+    this._button = this.querySelector('.btn');
+    this._disabled = /\bdisabled/.test(this._button.className);
+
+    this._button.addEventListener('click', () => {
+      this._showDropdown();
+    });
+
+    document.addEventListener('click', (event) => {
+      //close dropdown if clicked outside menu
+      if (event.target !== this && !this.contains(event.target)) {
+        this._clearDropdown();
+      }
+    });
+
+    document.addEventListener('keydown', (event) => {
+      if (/input|textarea/.test(event.target.tagName)) {
+        return;
+      }
+      if (this._disabled) {
+        return;
+      }
+      let keycode = event.keyCode ? event.keyCode : event.which;
+      let active = /\bopen/.test(this._button.parentNode.className);
+
+      //check if dropdown is open and keys other than 38,40,27,13 are not being used
+      if (active && /(38|40|27|13)/.test(keycode)) {
+        this._keyHandler(event, keycode);
+      }
+    });
+  }
+
+  /**
+   * Called when element's attribute value has changed
+   *
+   * @param {string} attrName The attribute name that has changed
+   * @param {string} oldValue The old attribute value
+   * @param {string} newValue The new attribute value
+   */
+  attributeChangedCallback(attrName, oldValue, newValue) {
+
+  }
+
+  /**
+   * Called when an instance of the element is created
+   */
+  createdCallback() {
+
+  }
+
+  /**
+   *Toggle the dropdown
+   */
+  toggle() {
+    this._showDropdown();
+  }
+
+  /**
+   * Open the dropdown
+   *
+   */
+  _showDropdown() {
+    if (/\bdisabled/.test(this._button.className)) {
+      return;
+    }
+    let active = this._button.parentNode.classList.contains('open');
+    if (!active) {
+      this._detectTouch();
+      this.dispatchEvent(new CustomEvent('show.bs.dropdown', {}));
+      this._button.focus();
+      this._button.setAttribute('aria-expanded', 'true');
+      this._button.parentNode.classList.toggle('open');
+      this.dispatchEvent(new CustomEvent('shown.bs.dropdown', {}));
+    }
+    if (active) {
+      this._clearDropdown();
+    }
+  }
+  /**
+   * Close the dropdown
+   *
+   */
+  _clearDropdown() {
+    let backdrop = this.querySelector('.dropdown-backdrop');
+    if (backdrop) {
+      backdrop.parentNode.removeChild(backdrop);
+    }
+    this.dispatchEvent(new CustomEvent('hide.bs.dropdown', {}));
+    this._button.setAttribute('aria-expanded', 'false');
+    this._button.parentNode.classList.remove('open');
+    this.dispatchEvent(new CustomEvent('hidden.bs.dropdown', {}));
+  }
+
+  /**
+   * Support for phone browser
+   *
+   */
+  _detectTouch() {
+    if ('ontouchstart' in document.documentElement) {
```

Comment:
It should be working fine now. There was an error because at first, I was using `<pf-dropdown class="dropdown">` but to keep bootstrap API structure intact I changed it to `<pf-dropdown><div class="dropdown"> .... </div></pf-dropdown>`